src/metabase/channel/email.clj

```diff
@@ -0,0 +1,106 @@
+(ns metabase.channel.email
+  (:require
+   [metabase.channel.core :as channel]
+   [metabase.email :as email]
+   [metabase.email.messages :as messages]
+   [metabase.query-processor.timezone :as qp.timezone]
+   [metabase.util.i18n :refer [trs]]
+   [metabase.util.malli :as mu]
+   [metabase.util.malli.schema :as ms]
+   [toucan2.core :as t2]))
+
+(channel/register! :channel/slack)
+
+(mu/defn defaulted-timezone :- :string
+  "Returns the timezone ID for the given `card`. Either the report timezone (if applicable) or the JVM timezone."
+  [card :- (ms/InstanceOf :model/Card)]
+  (or (some->> card :database_id (t2/select-one :model/Database :id) qp.timezone/results-timezone-id)
+      (qp.timezone/system-timezone-id)))
+
+(defn- construct-pulse-email [subject recipients message]
+  {:subject      subject
+   :recipients   recipients
+   :message-type :attachments
+   :message      message})
+
+(defn- send-emails!
+  [emails]
+  (doseq [{:keys [subject recipients message-type message]} emails]
+    (email/send-message-or-throw! {:subject      subject
+                                   :recipients   recipients
+                                   :message-type message-type
+                                   :message      message
+                                   :bcc?         (email/bcc-enabled?)})))
+
+(defn- recipients->emails
+  [recipients]
+  {:user-emails     (mapv (comp :email :user) (filter #(= :user (:kind %)) recipients))
+   :non-user-emails (mapv :recipient (filter #(= :external-email (:kind %)) recipients))})
+
+;; ------------------------------------------------------------------------------------------------;;
+;;                                           Alerts                                                ;;
+;; ------------------------------------------------------------------------------------------------;;
+
+(defn- alert-condition-type->description [condition-type]
+  (case (keyword condition-type)
+    :meets (trs "reached its goal")
+    :below (trs "gone below its goal")
+    :rows  (trs "results")))
```

Comment:
I would prefer the whole subject generated in that one `case`, translation will be easier and of better quality.